query/src/sql/planner/semantic/type_check.rs

```diff
@@ -1133,33 +1133,60 @@ impl<'a> TypeChecker<'a> {
                 };
                 Some(self.resolve_function(span, "database", &[&arg], None).await)
             }
+            "currentdatabase" => {
+                let arg = Expr::Literal {
+                    span: &[],
+                    lit: Literal::String(self.ctx.get_current_database()),
+                };
+                Some(self.resolve_function(span, "currentdatabase", &[&arg], None).await)
+            }
+            "current_database" => {
+                let arg = Expr::Literal {
+                    span: &[],
+                    lit: Literal::String(self.ctx.get_current_database()),
+                };
+                Some(self.resolve_function(span, "current_database", &[&arg], None).await)
+            }
             "version" => {
                 let arg = Expr::Literal {
                     span: &[],
                     lit: Literal::String(self.ctx.get_fuse_version()),
                 };
                 Some(self.resolve_function(span, "version", &[&arg], None).await)
             }
-            "current_user" => match self.ctx.get_current_user() {
+            "user" => match self.ctx.get_current_user() {
+                Ok(user) => {
+                    let arg = Expr::Literal {
+                        span: &[],
+                        lit: Literal::String(user.identity().to_string()),
+                    };
+                    Some(self.resolve_function(span, "user", &[&arg], None).await)
+                }
+                Err(e) => Some(Err(e)),
+            },
+            "currentuser" => match self.ctx.get_current_user() {
```

Comment:
I think we do not need `currentuser`.

current_user is for clickhouse
user is for mysql

Reply:
If not support this, it will panic. Maybe we can delete it in another PR (also delete it when planner = v1)